benchmarks/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ScopePassingTraceContextHookConfig.java

```diff
@@ -0,0 +1,133 @@
+
+import brave.Tracing;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.reactivestreams.Publisher;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.cloud.sleuth.instrument.async.TraceableScheduledExecutorService;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import reactor.core.CoreSubscriber;
+import reactor.core.Scannable;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.GroupedFlux;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Operators;
+import reactor.core.scheduler.Schedulers;
+import reactor.util.context.Context;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+
+/**
+ * Replace default of onEachOperatorHook({@link ScopePassingSpanSubscriber}) to the {@link ScopePassingTraceContextSubscriber}.
+ */
+@Configuration
+public class ScopePassingTraceContextHookConfig {
+	@Bean
+	public ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor hookRegisteringBeanDefinitionRegistryPostProcessor(ConfigurableApplicationContext context) {
+		return new ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor(context);
+	}
+
+}
+
+class ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor extends HookRegisteringBeanDefinitionRegistryPostProcessor {
+
+	private static final Log log = LogFactory.getLog(ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor.class);
+	private final ConfigurableApplicationContext context;
+
+	ScopePassingTraceContextHookRegisteringBeanDefinitionRegistryPostProcessor(ConfigurableApplicationContext context) {
+		super(context);
+		this.context = context;
+	}
+
+	@Override
+	void setupHooks( BeanFactory beanFactory) {
+		Hooks.onEachOperator(
+				TraceReactorAutoConfiguration.TraceReactorConfiguration.SLEUTH_TRACE_REACTOR_KEY,
+				scopePassingSpanOperator(this.context));
+		Schedulers.setFactory(factoryInstance(beanFactory));
+	}
+
+
+	private static <T> SpanSubscriptionProvider<T> scopePassingTraceContextSubscription(
+			ConfigurableApplicationContext beanFactory, Scannable scannable, CoreSubscriber<? super T> sub) {
+		return new SpanSubscriptionProvider<T>(
+				beanFactory,
+				sub,
+				sub.currentContext(),
+				scannable.name()) {
+			@Override
+			SpanSubscription newCoreSubscriber(Tracing tracing) {
+				return new ScopePassingTraceContextSubscriber(
+						sub,
+						sub != null ? sub.currentContext() : Context.empty(),
+						tracing);
+			}
+		};
+	}
+
+	private Schedulers.Factory factoryInstance(final BeanFactory beanFactory) {
+		return new Schedulers.Factory() {
```

Comment:
this looks like it could be a lambda? if so, make it one?

Reply:
It's count not be a lambda.